db/model/topics.model.js

```diff
@@ -44,14 +44,30 @@ exports.selectArticleById = (article_id) => {
         COUNT(comments.comment_id)::INT AS comment_count
       FROM articles
       LEFT JOIN comments ON articles.article_id = comments.article_id
+    `;
+    
+    const queryParams = [];
+    
+    if (topic) {
+      query += ` WHERE articles.topic = $1`;
+      queryParams.push(topic);
+    }
+    
+    query += `
       GROUP BY articles.article_id
       ORDER BY ${sort_by} ${order};
     `;
-  
-    const { rows } = await db.query(query);
+    if (topic) {
+      const { rows: topicRows } = await db.query("SELECT * FROM topics WHERE slug = $1", [topic]);
+      if (topicRows.length === 0) {
+        throw { status: 404, msg: "Topic not found" };
+      }
+    }
```

Comment:
You could extract the above out into it's own model and then use a Promise.all inside your controller calling both models. Have a look at the advanced errors notes!